### DISCUSSION TITLE
[RFC] Include taxes in `Order` undiscounted prices

### DISCUSSION THREAD

**Discussion by IKarbowiak** [saleor/saleor#14444] [RFC]
### Problem
The `undiscounted` prices don't include the tax rate, the `undiscounted net` and `undiscounted gross` price is the same.
The return prices should include the same tax rate as the discounted prices.

### General Assumptions
- If the prices are entered as gross (with tax included), calculate each line's undiscounted net using undiscounted gross and the tax rate
- If the prices are entered as net (without tax included),  calculate each line's undiscounted gross using undiscounted net and the tax rate

### ToDo
- Update `complete_checkout` method to create `OrderLines` with correct `undiscounted prices` that include the tax rate (tax calculation should be done based on the `tax rate`, in similar way as it's done for a flat rates).
- Ensure that `create_order_from_checkout` method also calculated the correct `undiscounted prices`.
- Make sure subtotals and totals are correct.
- Update `fetch_order_prices_if_expired` method - the undiscounted values should include taxes, the taxes on shipping need to be calculated too
- Ensure the `Order` `API` fields return the correct values for discounted and undiscounted fields
- Ensure that correct values are returned in the `Order` payload
- Add data migration for `Order` and `OrderLine` to update the `undiscounted` prices to include the tax.

**Comment by korycins** [comment]
> Ensure the Order API fields return the correct values for discounted and undiscounted fields

Do we want to do the same thing for a `CheckoutAPI`?  The checkout line type also has fields for undiscounted prices.

> Ensure that correct values are returned in the Order payload

In case if we will calculate undiscounted prices for checkout, we should also double-check the payload that we send for Checkout.


Please double-check whether adding or removing vouchers to checkout, as well as adding or removing OrderDiscounts/OrderLineDiscounts to Order, results in the correct amounts.

Additionally, please verify that the orderDiscount/OrderLineDiscount added to the order returns the correct amount when changing the quantity of the order line or adding a new line to the order.

**Comment by IKarbowiak** [comment]
@korycins it will be done only for `Order`.
